router/router.go

```diff
@@ -761,7 +761,7 @@ func (worker *workerT) processDestinationJobs() {
 				}
 				ch <- struct{}{}
 				timeTaken := time.Since(startedAt)
-				if respStatusCode != types.RouterTimedOutStatusCode {
+				if respStatusCode != types.RouterTimedOutStatusCode && respStatusCode != types.RouterUnMarshalErrorCode {
```

Comment:
Why is it bad to include these in the workspace latency calculation algorithm and it is not bad to include any other >500 response code?

Reply:
So Internal Status codes would mean that we have not made any attempt to send the Job to the destination, So this metric will reduce the latency value for the workspace giving us a false sense of faster destination

Reply:
ok, thanks. I was under the impression that `RouterTimedOutStatusCode` would increase it, but maybe I don't have a good understanding yet of under which conditions we fail with a `RouterTimedOutStatusCode`

Reply:
So whenever we pick some jobs from `router` we assign a time after which all those jobs should have been attempted. If we encounter any jobs from that batch which haven't been attempted yet we give this error `RouterTimedOutStatusCode` and simple drop the job